### PR TITLE
Add openhab2 with reverse proxy, remove auto letsencrypt

### PR DESCRIPTION
**Pull request by hestela**
Main issue i have is that when you try to unpack or cp the files into
the user's home directory the permissions get screwed up. chown -R didn't
work. For now I just switched to the openhab user and unpacked the
pre-built files.

Openhab2 does not support authentication at the moment so I am using
nginx basic auth. Considering adding fail2ban to monitor the nginx auth
logs. Seems that the habmin interface keeps asking for auth repeatedly.

Need to add a crontab entry to run certbot renew

will keep conf directory files here:
[hestela/openhab-conf](https://github.com/hestela/openhab-conf)

### CODE REVIEW

File: nix/openhab.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ config, pkgs, ... }:
+
+{
+  users.extraUsers.openhab = {
+    isNormalUser = true;
```
Comment (hestela) [inline nix/openhab.nix]: i should revisit this, documentation says it should be a system user

File: nix/openhab.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      jre
+      bash
+      gawk
+    ];
+    description = "The openHAB 2 Home Automation Bus Solution";
+    wantedBy = [ "multi-user.target" ];
+    after = [ "syslog.target" "network.target" ];
+    environment.HOME = "/home/openhab";
+    environment.USER = "openhab";
+    # TODO: fix manual unpack of zip to home
```
Comment (hestela) [inline nix/openhab.nix]: the nix builder cant extract to openhab's homedir

**Comment by hestela** [comment]
@acmcarther, idk how interested you are at smart home stuff. but this is automation for connected devices.  my co-worker uses it to schedule things like his sprinkler system, lights. z-wave motions sensors to run scripts, etc. openhab v2 if you are interested. the runtime is java, started by a /bin/bash script. i have anix expression for getting the zip, had some wierd problems i think i know what i could try to get it to work.

**Comment by acmcarther** [comment]
ACK:

I have [homeassistant configs](https://github.com/acmcarther/essentials/blob/master/nix/custom-nix-pkgs/homeassistant/default.nix) on my own box, but I've turned em off because keeping the dependencies up to date has been a royal PITA.

File: nix/server.nix [hunk 3]
```diff
@@ -117,8 +118,8 @@ in
   networking = {
     hostName = "quid"; # Define your hostname.
     hostId = "e39841f0";
-    firewall.allowedTCPPorts = [ 80 443 3000 8888 42063 8000 ];
-    firewall.allowedUDPPorts = [ 80 443 3000 8888 42063 8000 ];
+    firewall.allowedTCPPorts = [ 80 443 3000 8888 42063 8000 8080 ];
```
Comment (acmcarther) [inline nix/server.nix]: Is exposing this port actually necessary? nginx should be proxying from 80/443....

File: nix/development.nix [hunk 4]
```diff
@@ -51,6 +51,7 @@
       go
       go2nix
       godep
+      jre
```
Comment (acmcarther) [inline nix/development.nix]: You shouldn't need this dependency here, unless you actually want it as a system dep. Adding it under openhab2's config should work.

File: nix/nginx.nix [hunk 5]
```diff
@@ -103,6 +90,35 @@ in
               chunked_transfer_encoding off;
             }
         }
+        # OpenHab 2
+        server {
+            listen 443;
+            server_name oh2.easycashmoney.org;
+            auth_basic "OpenHab2";
+            auth_basic_user_file /var/www/.htpasswd;
```
Comment (acmcarther) [inline nix/nginx.nix]: Neat.

File: nix/server.nix (hunk 3 above)
Comment (hestela) [reply nix/server.nix]: For LAN access. one of the UIs they have doesnt work with nginx auth, no one has bothered to fix it....

File: nix/development.nix (hunk 4 above)
Comment (hestela) [reply nix/development.nix]: oh yeah, thats right. forgot i had it in the systemd unit

**Pull request merged**